kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -237,4 +257,84 @@ public void put(int priority, K key, V value) {
             return map.entrySet();
         }
     }
-}
+
+    private static Collection<URL> findKieConfUrls(ClassLoader cl) throws IOException {
+        List<URL> kieConfsUrls = new ArrayList<>();
+
+        Enumeration<URL> metaInfs = cl.getResources(CONF_FILE_FOLDER);
+        while (metaInfs.hasMoreElements()) {
+            URL metaInf = metaInfs.nextElement();
+            if (metaInf.getProtocol().startsWith("vfs")) {
+                // the kie.conf discovery mechanism doesn't work under OSGi
+                kieConfsUrls.clear();
+                break;
+            }
+
+            URLConnection con = metaInf.openConnection();
+            if (con instanceof JarURLConnection) {
+                collectKieConfsInJar(kieConfsUrls, metaInf, (JarURLConnection) con);
+            } else {
+                collectKieConfsInFile(kieConfsUrls, new File(metaInf.getFile()));
+            }
+        }
+
+        if (kieConfsUrls.isEmpty()) {
+            // no kie-conf found so fallback to the hardcoded lookup
+            kieConfsUrls = getKieConfsFromKnownModules(cl).collect(Collectors.toList());
+        } else {
+            // check if all discovered kie.conf file are in known modules
+            kieConfsUrls.stream().map(ServiceDiscoveryImpl::getModuleName)
+                    .filter(module -> Arrays.binarySearch(KIE_MODULES, module) < 0)
+                    .forEach(module -> log.warn("kie.conf file discovered for '" + module + "' but not listed among the known modules. This will not work under OSGi or JBoss vfs."));
```

Comment:
I think this should throw an exception: if this happens it means that a `kie.conf` has been found under the new path `META-INF/kie/` that is not listed by `KIE_MODULES` so it should never happen.
User has a reserved path to use while all our repositories should be updated (or keep the legacy path in the worst case scenario).